packages/xo-web/src/xo-app/backup-ng/new/index.js

```diff
@@ -521,16 +443,39 @@ export default [
 
         return getInitialState()
       },
-      setReportWhen: (_, { value }) => () => ({
-        reportWhen: value,
-      }),
-      setConcurrency: (_, concurrency) => (_, { job }) => ({
-        concurrency:
-          concurrency === undefined && job !== undefined ? '' : concurrency,
-      }),
-      setTimeout: (_, timeout) => (_, { job }) => ({
-        timeout: timeout === undefined && job !== undefined ? '' : timeout,
+      setGlobalSettings: (_, { name, value }) => ({
+        propSettings,
+        settings = propSettings,
+      }) => ({
+        settings: settings.setIn(['', name], value),
       }),
+      setReportWhen: ({ setGlobalSettings }, { value }) => () => {
+        setGlobalSettings({
+          name: 'reportWhen',
+          value,
+        })
+      },
+      setConcurrency: ({ setGlobalSettings }, value) => () => {
+        setGlobalSettings({
+          name: 'concurrency',
+          value,
+        })
+      },
+      setTimeout: ({ setGlobalSettings }, value) => () => {
+        setGlobalSettings({
+          name: 'timeout',
+          value: value && value * 1e3,
+        })
```

Comment:
You're not keeping the same behaviour for `concurrency` and `timeout`, which may break. Is it normal?